test/Microsoft.Azure.Relay.UnitTests/CustomClientWebSocketFactory.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Relay.UnitTests
+{
+    public class CustomClientWebSocketFactory : IClientWebSocketFactory
+    {
+        public bool IsCreatedCalled;
+
+        public CustomClientWebSocketFactory()
+        {
+            this.IsCreatedCalled = false;
+        }
+
+        public IClientWebSocket Create()
+        {
+            IsCreatedCalled = true;
+            return new CustomClientWebSocket(new ClientWebSocket());
+        }
+    }
+
+    // Custom websocket implementation.
+    class CustomClientWebSocket : IClientWebSocket
```

Comment:
>CustomClientWebSocket [](start = 10, length = 21)

Nit: I'd nest this class to avoid having multiple top level classes in a single C# file.

Reply:
done